### DISCUSSION TITLE
yarpmotorgui  dependencies

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#628] [Q&A] [answered]
what is the dependency of launching yarpmtorgui in icubsim, i succeed sometimes but fail too

**Comment by jimingda** [comment]
![2023-05-24 17-30-21 的屏幕截图](https://github.com/robotology/community/assets/7019692/0b6e207f-62cb-434e-b7ea-c68491654970)

**Comment by pattacini** [comment, marked as answer]
Hi @jimingda 

From the screenshot[^1] you posted, I can clearly see that you're trying to connect to `icub` rather than `icubSim`, hence the `yarpmotorgui` failure.

Try out this instead:
```console
yarpmotorgui --robot icubSim
```

[^1]: It's always better to post a file rather than a screenshot, for inspection purposes.

**Reply by jimingda** [reply]
Thanks, if icub is power on and connected to laptop, the command yarpmotorgui will connect to icub directly rather than icubsim?

I launch yarpserver and yarpmanager, and then try comman "yarpmotorgui --robot icubSim" and failed, as show in the log below.

[yarpmotorgui icubsim docker.log](https://github.com/robotology/community/files/11553331/yarpmotorgui.icubsim.docker.log)

**Reply by pattacini** [reply]
>Thanks, if icub is power on and connected to laptop, the command yarpmotorgui will connect to icub directly rather than icubsim?

The `yarpmotorgui` will try to connect to `icub` by default. The robot name can be overridden by means of the `--robot` command line option.

>I launch yarpserver and yarpmanager, and then try comman "yarpmotorgui --robot icubSim" and failed, as show in the log below.

You need to have the simulator up and running as well.

In fact, from the log we can see that there are no robot motor ports available as for example:
```console
[ERROR] |yarp.device.remote_controlboard| Problem connecting to /icubSim/head/rpc:i, is the remote device available?
[ERROR] |yarp.device.remote_controlboard| Problem connecting to /icubSim/head/command:i, is the remote device available?
[ERROR] |yarp.device.remote_controlboard| Problem connecting to /icubSim/head/stateExt:o, is the remote device available?
```

**Reply by jimingda** [reply]
`You need to have the simulator up and running as well.`
What does the simulator refer to, yarprobotinterface? I tried to change the yarprobotinterface's host from icub-head to localhost, but failed to launch it.

**Reply by pattacini** [reply]
It simply means that you have to make sure to launch Gazebo before launching `yarpmotorgui`.
I'm saying that because the error messages you reported above clearly indicate that the simulator motor ports were not available.

**Comment by jimingda** [comment]
Is there any place I can refer to the usages of these yarp commands, like parameters ?

**Reply by pattacini** [reply]
YARP is documented at https://yarp.it.

In particular, `yarpmotorgui` is documented at https://yarp.it/latest/group__yarpmotorgui.html.
